### PR TITLE
Overview3

### PR DESCRIPTION
**Pull request by EJLEE808**
This is big ugly. 
I apologize for this huge pull request. 

1. Redid my components with "new" 
2. Added "On Sale" functionality
3. Tried to add modal window.
4. Stopped to make a pull request before it got even bigger.

### CODE REVIEW

File: client/src/components/overview/components/newAmount.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import React from 'react'
```
Comment (PBandJaimie) [inline client/src/components/overview/components/newAmount.jsx:1]: Great variable names so far!!!!

File: client/src/components/overview/components/newButton.jsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+
+let NewButton = (props) => {
+  if (props.availableAmount === '0') {
+    return (
+      <div></div>
```
Comment (PBandJaimie) [inline client/src/components/overview/components/newButton.jsx:6]: I love how you are handling possible missing data with these `<div></div>s`. Something I need to do!

File: client/src/components/overview/components/newPhotos.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        height='150'
+        width='120'
+        key={i}
+        src={props.thumbnails[i].thumbnail_url}
+        onClick={e => {
+          e.preventDefault()
+          props.setMainPhoto(props.thumbnails[i].url)
+          props.setThumbnailIndex('' + i)
+        }}
+        alt='Thumbnail' ></img>
```
Comment (PBandJaimie) [inline client/src/components/overview/components/newPhotos.jsx:17]: Wonderful for accessibility!

File: client/src/components/overview/overview.jsx [hunk 4]
```diff
@@ -1,33 +1,121 @@
...
+import NewPhotos from './components/NewPhotos.jsx'
+import NewSize from './components/newSize.jsx'
+import NewAmount from './components/newAmount.jsx'
+import NewButton from './components/newButton.jsx'
+import Modal from './components/modal.jsx'
 
 class Overview extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
```
Comment (PBandJaimie) [inline client/src/components/overview/overview.jsx:12]: this is a very hefty state! impressive!

File: client/src/components/overview/overview.jsx [hunk 5]
```diff
@@ -1,33 +1,121 @@
...
 
-  render() {
-    let renderFeatures, renderStyles
-    if (this.state.info !== '') {
-      renderFeatures = <Features features={this.state.info.features} />
-      renderStyles = <Styles styles={this.state.styles.results} />
+
+  componentDidMount() {
+    // console.log('bigProps', this.props)
+    if (this.props === undefined) {
```
Comment (PBandJaimie) [inline client/src/components/overview/overview.jsx:94]: What happens if this.props === undefined? the block is blank.

File: client/src/components/overview/overview.jsx [hunk 6]
```diff
@@ -1,33 +1,121 @@
...
+        // chosenSize: this.props.styles.results[0].skus['941206'].size,
+        availableSizes: this.props.styles.results[0].skus,
+        // availableAmount: this.props.styles.results[0].skus['941206'].quantity,
+        // chosenAmount: '',
+        mainPhoto: this.props.styles.results[0].photos[0].url,
+        thumbnails: this.props.styles.results[0].photos,
+      })
     }
+  }
+  componentDidUpdate() {
```
Comment (PBandJaimie) [inline client/src/components/overview/overview.jsx:109]: Do you still need a componentDidUpdate() function here?

File: client/src/components/overview/overview.jsx [hunk 7]
```diff
@@ -38,13 +126,71 @@ class Overview extends React.Component {
...
+          chosenAmount={this.state.chosenAmount}
+          skuID={this.state.skuID}
+          showWarning={this.state.showWarning}
+          availableAmount={this.state.availableAmount}
+        />
       </div>
     );
   }
 }
 export default Overview;
```
Comment (PBandJaimie) [inline client/src/components/overview/overview.jsx:196]: This Overview file is huge, but more than that it is brilliant. I really love the way it is organized. Makes me want to redo all of my stuff... lol.

File: client/src/spec/overview.spec.js [hunk 8]
```diff
@@ -35,4 +35,10 @@ describe('Overview', () => {
     let tree = component.toJSON();
     expect(tree).toMatchSnapshot();
   });
-});
\ No newline at end of file
+
```
Comment (PBandJaimie) [inline client/src/spec/overview.spec.js:38]: Great first test! Clever to use Jest() function  in the Overview component.

File: webpack.config.js [hunk 9]
```diff
@@ -5,6 +5,7 @@ var DIST_DIR = path.join(__dirname, '/client/dist');
 
 module.exports = {
   entry: `${SRC_DIR}/index.js`,
+  cache: false,
```
Comment (PBandJaimie) [inline webpack.config.js:8]: 👍🏻

**Review by PBandJaimie — CHANGES_REQUESTED** [review]
@EJLEE808 run the command `git rm --cached coverage -r` and push back up with your console.logs/comments deleted and I'll approve!

File: client/dist/index.html [hunk 10]
```diff
@@ -4,6 +4,7 @@
     <title>Atelier</title>
     <link rel="stylesheet" type="text/css" href="styles.css">
     <link rel="stylesheet" type="text/css" href="./overviewCSS/overview.css">
+    <link rel="stylesheet" type="text/css" href="./overviewCSS/OVmodal.css">
```
[the remark this answers is not in the document]
Comment (a-nouraldean) [reply client/dist/index.html:7]: I do not think this a problem but we could test the speed?

[the remark this answers is not in the document]
Comment (EJLEE808) [reply client/dist/index.html:7]: Like before, we can copy paste it later, for development let's leave it as is

File: client/src/app.jsx [hunk 11]
```diff
@@ -102,7 +102,12 @@ class App extends React.Component {
...
   }
 
+  componentDidMount() {
+    this.getAllProductInfo('28212')
+  }
+
   render() {
+
     let overview;
     if (this.state.overview.id === undefined) {
```
[the remark this answers is not in the document]
Comment (EJLEE808) [reply client/src/app.jsx:112]: I don't know why you mention object, but this is conditional rendering I used.

File: client/src/components/overview/components/modal.jsx [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+              <button onClick={e => {console.log('click X')}}>X</button>
+              <h4 className="OVmodal-title">Title</h4>
+            </div>
+            <div className="OVmodal-body">
+              Modal Content
+              {thumbnails}
+            </div>
+            <div className="OVmodal-footer">
+              <button onClick={e => { props.setShowModal() }} className="OVmodal-button">Close</button>
+            </div>
```
[the remark this answers is not in the document]
Comment (EJLEE808) [reply client/src/components/overview/components/modal.jsx:24]: RIGHT!? I loved this idea so our CSS won't overlap

File: client/src/components/overview/overview.jsx (hunk 5 above)
Comment (EJLEE808) [reply client/src/components/overview/overview.jsx:94]: Nothing happens, it used to have something there, but I deleted it and didn't refactor so it looks just empty.

File: client/src/components/overview/overview.jsx (hunk 6 above)
Comment (EJLEE808) [reply client/src/components/overview/overview.jsx:109]: I can just delete it next time.

File: client/src/components/overview/overview.jsx (hunk 7 above)
Comment (EJLEE808) [reply client/src/components/overview/overview.jsx:196]: I'm glad you think that way! I don't think it's 100% recommended during job hunts, but this philosophy really has some good trade offs to offer.

**Pull request merged**